core/templates/dev/head/pages/library/Library.js

```diff
@@ -198,15 +221,19 @@ oppia.controller('Library', [
       }
     };
 
-    $scope.showAllResultsForCategories = function(categories) {
-      var selectedCategories = {};
-      for (i = 0; i < categories.length; i++) {
-        selectedCategories[categories[i]] = true;
-      }
+    $scope.showAllResultsForCategories = function(categories, fullResultsUrl) {
```

Comment:
Maybe rename to showFullResultsPage, since some of these pages don't correspond to categories.

Also add docstring to explain what this function really does (and, in particular, that if fullResultsUrl is supplied, categories is ignored). That's not going to be obvious to a casual reader.